test/create-app.js

```diff
@@ -516,4 +517,21 @@ describe('create-app', () => {
         .catch(done);
     });
   });
+  describe('calling a callback method', () => {
+    let results;
+    before(() =>
```

Comment:
this is a personal preference thing, but I don't love the pattern of using a very localized `before`. I think it's more clear to have a function that returns the results. I _think_ mocha on this version is cool with `async // await` but if not, it's fine to leave as is

Reply:
Thanks for this feedback.

Are you saying you would prefer if each it block called the method and then asserted against the results?

I'm generally against that as it slows down the tests unless you bung a bunch of asserts in the same it block, which then obscures failures and foes you to have to dive into the it block to see what's going on.

I've always followed a describe per scenario, and we're pushing that in cli apps (https://zapier.quip.com/x4SEApNJLlRe)

by which I only mean to add a second opinion, I'm happy to follow the standard laid out before me for this repository, just want to check I'm understanding correctly.

Reply:
Sorry, I could have been more clear. I was picturing this:

```js
// instead of
describe(...
  let results
  before(() => someFunc().then(output => {results = output}))
  // it() using `results`

// take away the before
describe('taking a callback method', () => {
  const genResults = () => app(createTestInput('resources...'))

  it('returns a C...', async () => {
    const results = await genResults()
    results.status.should.eql('CALLBACK'))
  })

  it('returns m...', async () => {
    const results = await genResults()
    results.status.should.eql('CALLBACK'))
  })
})
```

this version makes it a more clear where `results` are coming from. I'm not sure if there are performance considerations (it feels like two pretty similar ways to do the same thing) but if there are, faster is probably better.

Reply:
@xavdid just to complete the circle on this conversation.  `const results = await genResults()` is called twice in your example which means that function will execute two discrete times, in the before method it gets called once (as the setup for the tests) and then the tests get called on the output from calling it once. Hence why I fallback to using the `before` pattern. I'm ok with changing it but it is more expensive.

Reply:
totally. I think in this case, there's a case either way (and I'm fine for you to leave as is). It would be more of an issue if we were worried about keeping `results` fresh for each `it`, since with this setup, the tests could affect each other. Agree that it doesn't matter here though.